pybamm/models/lithium_ion/spm.py

```diff
@@ -80,10 +95,78 @@ def __init__(self):
         electrode_vars = electrode_model.get_explicit_leading_order(self.variables)
         self.variables.update(electrode_vars)
 
+        "-----------------------------------------------------------------------------"
+        "Boundary conditions"
+        if self.options["bc_options"]["dimensionality"] == 2:
+            current_collector_model = pybamm.current_collector.OhmTwoDimensional(param)
+            #current_collector_model.set_uniform_current(self.variables)
+            current_collector_model.set_algebraic_system_spm(self.variables)
+            self.update(current_collector_model)
+
         # Cut-off voltage
-        voltage = self.variables["Terminal voltage"]
-        self.events.append(voltage - param.voltage_low_cut)
+        # TO DO: get terminal voltage in 2D
+        if self.options["bc_options"]["dimensionality"] == 0:
+            voltage = self.variables["Terminal voltage"]
+            self.events.append(voltage - param.voltage_low_cut)
 
     @property
     def default_geometry(self):
-        return pybamm.Geometry("1D macro", "1D micro")
+        dimensionality = self.options["bc_options"]["dimensionality"]
+        if dimensionality == 0:
+            return pybamm.Geometry("1D macro", "1D micro")
+        elif dimensionality == 1:
+            return pybamm.Geometry("1+1D macro", "(1+0)+1D micro")
+        elif dimensionality == 2:
+            return pybamm.Geometry("2+1D macro", "(2+0)+1D micro")
+
+    @property
+    def default_submesh_types(self):
+        dimensionality = self.options["bc_options"]["dimensionality"]
+        if dimensionality in [0, 1]:
+            return {
+                "negative electrode": pybamm.Uniform1DSubMesh,
+                "separator": pybamm.Uniform1DSubMesh,
+                "positive electrode": pybamm.Uniform1DSubMesh,
+                "negative particle": pybamm.Uniform1DSubMesh,
+                "positive particle": pybamm.Uniform1DSubMesh,
+                "current collector": pybamm.Uniform1DSubMesh,
+            }
+        elif dimensionality == 2:
+            return {
+                "negative electrode": pybamm.Uniform1DSubMesh,
+                "separator": pybamm.Uniform1DSubMesh,
+                "positive electrode": pybamm.Uniform1DSubMesh,
+                "negative particle": pybamm.Uniform1DSubMesh,
+                "positive particle": pybamm.Uniform1DSubMesh,
+                "current collector": pybamm.FenicsMesh2D,
+            }
```

Comment:
note: could have a `base_submeshes` dict that is common to both cases, and then only need to update the current collector mesh (to reduce number of lines)